### ISSUE TITLE
Model details

### ISSUE THREAD

**Issue by lambdaofgod** [mahnazkoupaee/WikiHow-Dataset#5]
Could you share details - implementations, hyperparameters for models that you described in your paper? I'm trying to replicate your results using Seq2Seq framework.

**Comment by mahnazkoupaee** [comment]
The encoder is limited to 400 tokens, and articles with more than 400 tokens are truncated, and the decoder is limited to produce 100 token summaries. All hidden layers are of size 256. The word embeddings are 128 dimensional and are learned in the training process. A 50k vocabulary size is used. The summarization model is trained with a batch size of 16 and Adagrad optimizer with learning rate 0.15, and gradient clipping with a maximum gradient norm of 2 and all the parameters are tuned on the validation set.

**Comment by lambdaofgod** [comment]
What is the depth of your network? What are optimization parameters (most Seq2Seq frameworks have different default parameters for optimizers, for example for lowering learning rate on plateau). Also did you use early stopping criterion?

**Comment by mahnazkoupaee** [comment]
The learning rate is fixed during the training without any changes. We use the loss on the validation set to implement early stopping. 
At test time our summaries are produced using beam search with beam size 4. 
The encoder is a single-layer bidirectional LSTM and the decoder is a single-layer unidirectional LSTM. 
Stanford CoreNLP is used to tokenize the data. 

The implementation of the pointer-generator network (https://github.com/abisee/pointer-generator) and the way they process the CNN/Daily Mail data (https://github.com/abisee/cnn-dailymail) can help you with your implementations.
